content/post/2025-06-19-open-files-folders-websites/index.Rmd

```diff
@@ -0,0 +1,154 @@
+
+One thing in particular that Hannah showed me that looks cool is the use of a [project launcher](https://positron.posit.co/rstudio-rproj-file.html#use-an-application-launcher).
+
+Now, I myself still use the terminal. 
+For instance if I clone a repository[^usethis], I'll then open it with the `positron` command.
+
+Another way to open projects I use a lot is, on Positron as on Rstudio, clicking to the recent projects -- when the IDE is already up and running.
+Unfortunately, I created and [currently](https://stateofther.netlify.app/post/saperlipopette/) [often](https://www.meetup.com/rbuenosaires/events/308338205/) use a [package that creates throwaway folders for practicing Git](https://docs.ropensci.org/saperlipopette/) which makes the list of recent projects utter rubbish. 🙃
+
+Within saperlipopette itself, to create and open the exercice folder on behalf of the user I use `usethis::create_project()`, which handily opens it in Positron when run in Positron and RStudio when run in RStudio.
+
+[^usethis]: I know I could use `usethis::create_from_github()` and sometimes do but not always... Maybe the message of this post is that my workflows are a big mess.
+
+## Open an URL in the browser: `browseURL()` and {cli}
+
+Sometimes you want the user of some code of yours to go admire or read a web page.
+
+To open an URL in the default browser, you can use `utils::browseURL()`.
+In all of the `usethis::browse_` functions, `utils::browseURL()` is what's used under the hood.
+For instance:
+
+```{r, eval=FALSE}
+browseURL("https://masalmon.eu/post")
+```
+
+To make it easy to open an URL from a message, you can use the [URL class](https://cli.r-lib.org/reference/inline-markup.html#classes) of the cli package.
+For instance:
+
+```{r, eval=FALSE}
+cli::cli_alert_info("Go read that blog! {.url https://masalmon.eu/post}!")
+```
+
+When getting that message, the user will simply have to click on the link to follow it (after trusting the domain in Positron).
+
+## Open only in interactive sessions: `interactive()`, `rlang::is_interactive()`
+
+If you use `utils::file.edit()` or `utils::browseURL()` in your code, you need to ensure the session is interactive.
+
+Either
+
+```{r, eval=FALSE}
+if (interactive()) {
+  file.edit("config.toml")
+}
+```
+
+Or
+
+
+```{r, eval=FALSE}
+if (rlang::is_interactive()) {
+  file.edit("config.toml")
+}
+```
+
+Compared to `interactive()`, `rlang::is_interactive()` also checks whether knitr or testthat are in progress, and provides an escape hatch through the `rlang_interactive` option.
+
+All of usethis functions that try to open or browse something on behalf of the user behave differently based on `rlang::is_interactive()`.
+[`usethis::edit_file()`'s source](https://github.com/r-lib/usethis/blob/4aa55e72ccca131df2d98fcd84fff66724d6250a/R/edit.R#L20),
+[`usethis::view_url()`'s source](https://github.com/r-lib/usethis/blob/4aa55e72ccca131df2d98fcd84fff66724d6250a/R/helpers.R#L107).
```

Comment:
Are these supposed to be bullet points? Right now, they are sort of just squished there.